SimonAwiti/API-endpoints
# #160713184-configured with the database, with reviews.

## What this PR does
* Connection of the application to the database.
* Implementing blueprints as advised by the previous LFA
* Considering the reviews by the LFA

## The following are API endpoints enabling one to: 
* Place a new order for food.
* Get a list of orders.
* Fetch a specific order.
* Update the order status.
## Here is a list of the functioning endpoints

| EndPoint                | Functionality        | Routes            |
| :---                    |     :---:            | :---              |
| GET /orders             | Get all orders       | /api/v2/orders    |
| GET /orders/<orderId>   | Get order by order id| /api/v2/orders/id |
| POST /orders            | Adds an order        | /api/v2/orders    |
| PUT /orders             | Edit orders          | /api/v2/orders/id |
| POST /users             | register user         | /api/v2/user |
  
## Testing the endpoints

* Install python then using pip instal .. install flask
* clone the repo
* Ensure that postman is installed
* From your terminal locate the repo and run: python run.py
* open postman and test the endpoints
* Use unittest to run the the tests
